core/panic.c

```diff
@@ -36,6 +36,11 @@
 #include "ps.h"
 #endif
 
+/* If a device is flashed over USB bootloader, try to launch
+ * the bootloader again on crash so the user can re-flash it.
+ */
+#include "usb_board_reset.h"
```

Comment:
Looking at this change a second time, I think we should do the following:

```suggestion
#if defined(DEVELHELP) && defined(MODULE_USB_BOARD_RESET)
#include "usb_board_reset.h"
#endif
```
You can squash directly.

Reply:
I had it like that before, but then I can't use

```C
    if (IS_USED(MODULE_USB_BOARD_RESET)) {
        usb_board_reset_in_bootloader();
    } else {
        pm_off();
    }
```

later on and would have to do

```C
#ifdef MODULE_USB_BOARD_RESET
    usb_board_reset_in_bootloader();
#else 
    pm_off();
#endif
```

instead.

It won't make a difference either way, the C version might be slightly nicer to look at than the preprocessor one.

Reply:
I'm fine with the preprocessor version. With the `#ifdef/endif` around the `#include`, we ensure that potential future changes in `usb_board_reset.h` won't have any unwanted effect here. And this file is already full of `#ifdef` anyway.
Also note that we don't necessarily need to use `IS_USED` all the time. By the way, it's use is not cosmetic, it's just that is ensures the compiler will process all lines. Here with the preprocessor version, all paths will be covered anyway by Murdock in different builds (some boards are using usb_board_reset and others don't).